chain/chunks/src/lib.rs

```diff
@@ -1018,228 +1018,224 @@ impl ShardsManager {
     ) -> (std::time::Instant, &'static str, Option<PartialEncodedChunkResponseMsg>) {
         // Try getting data from in-memory cache.
         let started = Instant::now();
-        if let Some(entry) = self.encoded_chunks.get(&request.chunk_hash) {
-            let response = Self::prepare_partial_encoded_chunk_response_from_cache(request, entry);
-            return (started, "cache", response);
+        let PartialEncodedChunkRequestMsg { chunk_hash, part_ords, mut tracking_shards } = request;
+        let mut part_ords = part_ords.into_iter().collect::<HashSet<_>>();
+        if part_ords.is_empty() && tracking_shards.is_empty() {
+            return (started, "none", None);
+        }
+        let mut response = PartialEncodedChunkResponseMsg {
```

Comment:
What about when we fully reconstruct a chunk, we also store all the chunk parts in self.encoded_chunks. In other words. we keep the content of entries in encoded_chunks and entries in column PartialEncodedChunk consistent. This way, we don't have to perform the lookup from storage if the chunk has an entry in self.encoded_chunk.

Reply:
Isn't that still kind of a problem if the node restarted, and received some parts from the network but not others, while a full chunk is encoded on disk?